### ISSUE TITLE
Bug: [no-unused-vars] false positive when using multiple `infer` on type parameters

### ISSUE THREAD

**Issue by marcalexiei** [typescript-eslint/typescript-eslint#10946]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&fileType=.tsx&code=JYOwLgpgTgZghgYwgAgMIHsC2AHANhADwBUBPbCAHgAUAaZI5QyEAEwGdkAJIgWQBkAovkwRwAPmTIA3gChJ2KOmxsAXMioBuOcjBqiMgL4yZYMilLkqcKHBGQoAgmBsIw6KBSIWIEgLzavM0YnUXY0LDxCbwpQGGhkAH06WPiAETgwOAkAfmR0zOQ1EAgAN2gtY0JsdzAdIIBBZF96MysbO2hHZ0Q3DwwcfGIzCilkBFw4NjYAOVsINTZnUABzZAM6bn4hCBFxMWMgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
interface ComplexType<P, T extends HTMLElement>  {
  props: P;
  t: T
}

type TypeParameterExtractor<TType> =
  TType extends ComplexType<infer _, infer Data> ? Data : never;


export type A = TypeParameterExtractor<ComplexType<{ className: string }, HTMLElement>>
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

In the example, `TypeParameterExtractor` is used to extract the second type parameter from `ComplexType`. 
To achieve this, we are inferring the following two type parameters:

1. `_`
2. `TElement`

However, `@typescript-eslint/no-unused-vars` flags the first parameter as unused, it is necessary for inferring the second one.

### Actual Result

`_` parameter should not be flagged as unused.

**Comment by bradzacher** [comment]
The rule is right -- the `infer _` is unused and is unnecessary.
[For example you could replace it with `any` and the type would work the same](https://typescript-eslint.io/play/#ts=5.8.2&fileType=.tsx&code=JYOwLgpgTgZghgYwgAgMIHsC2AHANhADwBUBPbCAHgAUAaZI5QyEAEwGdkAJIgWQBkAovkwRwAPmTIA3gChJ2KOmxsAXMioBuOcjBqiMgL4yZYMilLkqcKHBGQoAgmBsIw6KAEYKRCxAkBebR8zRidRdjQsPEJfClAYaGQAfTp4xIARODA4CQB%2BZEzs5DUQCAA3aC1TcnozKxs7aEdnRDcoACZvXwCg31DmCIwcfGIzCjgQElSQBKgCrJzkfMK4YuRSiqgtY0JsdzAdEIBBD2R-WstrWwh7Zpc2ryHo0fIKKWQEXDg2NgA5a7UbGcoAA5sgDHRuPwhBAROIxDJdvtDjUju0zhcIPVrrcnPd3J0niNYu9Pt8-gDkECoKDwZDeIJhKIwGIETIgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

**Comment by marcalexiei** [comment]
I haven't think you could pass a predefined type, instead of using `infer _` 😅.

Thanks for the quick review!
